content/6-network-security/4-fully-private-cluster/3-deploy-2048/index.en.md

```diff
@@ -0,0 +1,102 @@
+---
+title : "Deploy a Sample Application"
+weight : 152
+---
+
+Deploy the game 2048 as a sample application to verify that the AWS Load Balancer Controller creates an AWS ALB as a result of the ingress object. Complete the steps for the type of subnet you're deploying to.
+
+Download the manifest.
+
+
+```bash
+curl -O https://raw.githubusercontent.com/kubernetes-sigs/aws-load-balancer-controller/v2.5.4/docs/examples/2048/2048_full.yaml
+```
+
+Since your nodes don't have access to the Amazon ECR Public image repository, then you need to pull the following container image and push it to a repository that your nodes have access to.
+
+Pull image from Public registry
+
+
+```bash
+docker pull public.ecr.aws/l6m2t8p7/docker-2048:latest
+```
+
+
+::::expand{header="Check Output"}
+```
+latest: Pulling from l6m2t8p7/docker-2048
+23d07b917726: Pull complete 
+75b73619860e: Pull complete 
+Digest: sha256:afc96bdad819bfac184a6e9a90096b68583cf5977e66fa985143bde37e847a50
+Status: Downloaded newer image for public.ecr.aws/l6m2t8p7/docker-2048:latest
+public.ecr.aws/l6m2t8p7/docker-2048:latest
+```
+::::
+
+Get the image id for the pulled image
+
+
+```bash
+IMAGE_ID=$(docker image public.ecr.aws/l6m2t8p7/docker-2048:latest -q)
+```
+
+
+::::expand{header="Check Output"}
+```
+eb0a3a80a5dd
+```
+::::
+
+Tag the image that you pulled with your registry, repository, and tag. Use the IMAGE_ID variable as image id from the previous command
+
+```bash
+docker tag $IMAGE_ID $ACCOUNT_ID.dkr.ecr.ap-south-1.amazonaws.com/docker-2048
+```
+
+
+Authenticate to your Private ECR registry.
+
+
+```bash
+aws ecr get-login-password —region $AWS_REGION | sudo docker login —username AWS --password-stdin $ACCOUNT_ID.dkr.ecr.ap-south-1.amazonaws.com
+```
+
+
+Push the image to the Private ECR repository
+
+
+```bash
+docker push $ACCOUNT_ID.dkr.ecr.$AWS_REGION.amazonaws.com/docker-2048
+```
+
+
+- Edit the file and find the line that says alb.ingress.kubernetes.io/scheme: internet-facing.
+- Change internet-facing to internal and save the file.
+- Also change the public image repository url "image: public.ecr.aws/l6m2t8p7/docker-2048:latest" to "image:$ACCOUNT_ID.dkr.ecr.ap-south-1.amazonaws.com/docker-2048:latest"
```

Comment:
Remove :latest as it create an error in workshop studio : 
<img width="562" alt="image" src="https://github.com/user-attachments/assets/e363458c-87c0-477f-aaa9-5a5b46367a26">


```suggestion
- Also change the public image repository url "image: public.ecr.aws/l6m2t8p7/docker-2048:" to "image:$ACCOUNT_ID.dkr.ecr.ap-south-1.amazonaws.com/docker-2048"
```

Reply:
corrected